ml-agents/mlagents/trainers/torch/components/reward_providers/extrinsic_reward_provider.py

```diff
@@ -5,11 +5,42 @@
 from mlagents.trainers.torch.components.reward_providers.base_reward_provider import (
     BaseRewardProvider,
 )
+from mlagents_envs.base_env import BehaviorSpec
+from mlagents.trainers.settings import RewardSignalSettings
 
 
 class ExtrinsicRewardProvider(BaseRewardProvider):
+    """
+    Evaluates extrinsic reward. For single-agent, this equals the individual reward
+    given to the agent. For the POCA algorithm, we want not only the individual reward
+    but also the team and the individual rewards of the other agents.
+    """
+
+    def __init__(self, specs: BehaviorSpec, settings: RewardSignalSettings) -> None:
+        super().__init__(specs, settings)
+        self.add_groupmate_rewards = False
+
     def evaluate(self, mini_batch: AgentBuffer) -> np.ndarray:
-        return np.array(mini_batch[BufferKey.ENVIRONMENT_REWARDS], dtype=np.float32)
+        indiv_rewards = np.array(
+            mini_batch[BufferKey.ENVIRONMENT_REWARDS], dtype=np.float32
+        )
+        total_rewards = indiv_rewards
+        if (
+            BufferKey.GROUPMATE_REWARDS in mini_batch
+            and BufferKey.GROUP_REWARD in mini_batch
+        ):
+            if self.add_groupmate_rewards:
```

Comment:
Invert these 2 ifs. No need to check the first one if there are no groumaterewards

Reply:
These if conditions could be better:

if self.add_groupmate_rewards and BufferKey.GROUPMATE_REWARDS in mini_batch : do the groupmate reward
if BufferKey.GROUP_REWARD in mini_batch : Do the group reward

Reply:
Updated